custom_components/adaptive_lighting/switch.py

```diff
@@ -239,53 +239,69 @@ def _split_service_data(service_data, adapt_brightness, adapt_color):
     return service_datas
 
 
-def _find_switch_with_any_of_lights(
-    hass: HomeAssistant,
-    lights: list[str],
-    service_call: ServiceCall,
-) -> AdaptiveSwitch:
-    """Find the switch that controls the lights in 'lights'."""
+def _get_switches_with_lights(
+    hass: HomeAssistant, lights: list[str], is_on: bool
+) -> list[AdaptiveSwitch]:
     config_entries = hass.config_entries.async_entries(DOMAIN)
     data = hass.data[DOMAIN]
-    switches = {}
+    switches = []
     for config in config_entries:
-        # this check is necessary as there seems to always be an extra config
-        # entry that doesn't contain any data. I believe this happens when the
-        # integration exists, but is disabled by the user in HASS.
```

Comment:
Could we leave these comments here so it's apparent why we're making the check? Or at least shorten.

Reply:
I added another comment.